Sources/SpeziOnboarding/OnboardingFlow/OnboardingNavigationPath.swift

```diff
@@ -0,0 +1,240 @@
+    public func append(_ onboardingStepType: any View.Type) {
+        let onboardingStepIdentifier = OnboardingStepIdentifier(fromType: onboardingStepType)
+        guard onboardingSteps.keys.contains(onboardingStepIdentifier) else {
+            print("""
+            "Warning: Invocation of `OnboardingNavigationPath.append(_:)` with an Onboarding view
+            that is not delineated in the `OnboardingStack`. Navigation action is void."
+            """)
+            return
+        }
+        
+        appendToInternalNavigationPath(of: onboardingStepIdentifier)
+    }
+    
+    
+    /// An invocation of this function moves the internal `NavigationPath` of the ``OnboardingNavigationPath`` to the passed custom onboarding `View` instance. Keep in mind that this custom `View` does not have to be declared within the ``OnboardingStack``. Resulting from that, the internal state of the ``OnboardingNavigationPath`` is still referencing to the last regular `OnboardingStep`.
+    /// This function is closly related to ``append(customViewInit:)``.
+    ///
+    /// - Parameters:
+    ///   - customView: A custom onboarding `View` instance that should be shown next in the onboarding flow. It isn't required to declare this view within the ``OnboardingStack``.
+    public func append(customView: any View) {
+        let customOnboardingStepIdentifier = OnboardingStepIdentifier(fromView: customView, custom: true)
+        customOnboardingSteps[customOnboardingStepIdentifier] = customView
+        
+        appendToInternalNavigationPath(of: customOnboardingStepIdentifier)
+    }
+    
+    
+    /// An invocation of this function moves the internal `NavigationPath` of the ``OnboardingNavigationPath`` to the passed custom onboarding `View` initializer. Keep in mind that this custom `View` does not have to be declared within the ``OnboardingStack``. Resulting from that, the internal state of the ``OnboardingNavigationPath`` is still referencing to the last regular `OnboardingStep`.
+    /// This function is closly related to ``append(customView:)``.
+    ///
+    /// - Parameters:
+    ///   - customViewInit: A custom onboarding `View` initializer that creates a `View` shown next in the onboarding flow. It isn't required to declare this view within the ``OnboardingStack``.
+    public func append(customViewInit: @MainActor () -> any View) {
+        let customView = customViewInit()
+        let customOnboardingStepIdentifier = OnboardingStepIdentifier(fromView: customView, custom: true)
+        customOnboardingSteps[customOnboardingStepIdentifier] = customView
+
+        appendToInternalNavigationPath(of: customOnboardingStepIdentifier)
+    }
+    
+    
+    /// Removes the last element on top of the internal `NavigationPath` of the ``OnboardingNavigationPath``, meaning one is able to manually move backwards within the onboarding navigation flow.
+    public func removeLast() {
+        Task { @MainActor in
```

Comment:
One approach to enforce that all accesses to the `path` are on the `@MainActor` using an annotation right on the `path`: 
```swift
@MainActor @Published var path = NavigationPath()
```

Reply:
@PSchmiedmayer Thanks for the tip! I could remove the `@MainActor in` from the `Task`, however, I still need the `Task` itself, otherwise Xcode is complaining about background thread publishes. Actually not 100% sure why this is needed (do we call the methods from outside the new Swift concurrency context? -> I don't think so?), but apparently it is.

Reply:
Hmm, yes, that could be the case. The change would at least enforce any future change to get a compiler warning if they don't use `@MainActor` which is an important safeguard against future regressions 👍